site/documentation/how-does-opa-work/index.md

```diff
@@ -0,0 +1,512 @@
+---
+nav_id: MAIN_DOCUMENTATION
+doc_id: HOW_DOES_OPA_WORK
+layout: documentation
+
+title: How Does OPA Work?
+---
+
+{% contentfor header %}
+
+# How Does OPA Work?
+
+OPA is a full-featured policy engine that runs as a host-local daemon alongside your service. You can think of it as a concierge for your service who can answer detailed questions on behalf of your users to meet their specific needs.
+
+{% endcontentfor %}
+
+{% contentfor body %}
+
+## Overview
+
+OPA’s RESTful APIs use JSON over HTTP so you and your users can integrate OPA with any programming language. At a high level, integrating OPA into your service involves:
+
+  * Deploying OPA alongside your service
+  * Pushing relevant data about your service’s state into OPA’s document store
+  * Offloading some or all decision-making to OPA by querying it
+
+When your service is integrated with OPA, your users will be able author and deploy custom policies that control the behavior of your service’s policy-enabled features. Furthermore, users can publish data to OPA that is not available to your service about their own deployment context.
+
+In the future, both your service and its users will be able to register for, and react to, notifications triggered when OPA detects a policy-relevant change.
+
+## Deployment
+
+Unless you embed OPA as a Go library, you will deploy it alongside your service – either directly as an operating system daemon or inside a container. In this way, transactions will have low latency and availability will be determined through shared fate with your service.
+
+When OPA starts for the first time, it will not contain any policies or data. Policies and data can be added, removed, and modified at any time. For example: by deployment automation software or your service as it is deployed, by your service during an upgrade, or by administrators as needed.
+
+## Data and Policies
+
+The primary unit of data in OPA is a document, which is similar to a JSON value. Documents typically correspond to single, self-contained objects and are capable of representing both primitive types (strings, numbers, booleans, and null) as well as structured types (objects, and arrays). Documents are created, read, updated, and deleted via OPA’s RESTful HTTP APIs.
+
+{% img '{{assets["data-model-dependencies.svg"].logical_path}}' width:'720' %}
+
+### Base Documents
+
+So-called base documents contain static, structured data stored in memory and optionally saved to disk for resiliency. Your service will publish and update base documents in order to describe its current state, and your users can do the same to include relevant data about the state of their own deployment context.
+
+Base documents are published and updated using OPA’s Data API. For example, the following request publishes a list of servers to OPA:
+
+```http
+PATCH /v1/data/servers HTTP/1.1
```

Comment:
Other examples are using https://example.com/ in the URL. Either way is fine by me.